app/src/main/java/com/infomaniak/mail/ui/MainActivity.kt

```diff
@@ -377,8 +381,10 @@ class MainActivity : BaseActivity() {
     }
 
     private fun setupSnackBar() {
+
         fun getAnchor(): View? = when (navController.currentDestination?.id) {
-            R.id.threadListFragment -> findViewById(R.id.newMessageFab)
+            R.id.threadListFragment -> (currentFragment as? ThreadListFragment)?.getAnchor()
+            R.id.searchFragment -> (currentFragment as? SearchFragment)?.getAnchor()
```

Comment:
as threadListFragment and searchFragment are both TwoPaneFragment, we could regroup this with an abstract getAnchor() in TwoPaneFragment so we only have 1 case